Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp

```diff
@@ -814,6 +814,14 @@ void HTMLInputElement::create_text_input_shadow_tree()
     update_placeholder_visibility();
 
     if (type_state() == TypeAttributeState::Number) {
+        auto dispatch_input_event = [this]() {
```

Comment:
This doesn't seem safe - what protects `this` from GC?

Can we do this from `HTMLInputElement::step_up_or_down` instead? And when doing so, reference the spec link https://html.spec.whatwg.org/multipage/input.html#common-input-element-events, which states:
> For input elements without a defined input activation behavior ... then when the user changes the element's value, the user agent must queue an element task on the user interaction task source given the input element to fire an event named input at the input element

Reply:
We can't do this from `HTMLInputElement::step_up_or_down` because that method can be called from JS and the input event should only be fired on user interaction.

There were several places in this file that do this exact thing, so I've refactored it into its own function and added some explanatory spec text. I assume I'm fine to pass `[this]` into `queue_an_element_task` directly, because the function is immediately wrapped in a `HeapFunction`.

Reply:
> the input event should only be fired on user interaction

Ah, makes sense

> I assume I'm fine to pass [this] into queue_an_element_task directly, because the function is immediately wrapped in a HeapFunction.

Yep, exactly